src/h3lib/lib/polyfill.c

```diff
@@ -0,0 +1,422 @@
+/*
+ * Copyright 2016-2021 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file polyfill.c
+ * @brief   Functions relating to the cell-to-polygon algorithm
+ */
+
+#include "polyfill.h"
+
+#include <string.h>
+
+#include "alloc.h"
+#include "coordijk.h"
+#include "h3Assert.h"
+#include "h3Index.h"
+#include "polygon.h"
+
+// Factor by which to scale the cell bounding box to include all children.
+// This was determined empirically by finding the smallest factor that
+// passed exhaustive tests.
+#define CHILD_SCALE_FACTOR 1.4
+
+static const LatLng NORTH_POLE = {M_PI_2, 0};
+static const LatLng SOUTH_POLE = {-M_PI_2, 0};
+
+/**
+ * For a given cell, return its bounding box. If coverChildren is true, the bbox
+ * will be guaranteed to contain its children at any finer resolution. Note that
+ * in this case no guarantee is provided as to the level of accuracy, and the
+ * bounding box may have a significant margin of error.
+ * @param cell Cell to calculate bbox for
+ * @param out  BBox to hold output
+ * @param coverChildren Whether the bounding box should cover all children
+ */
+H3Error cellToBBox(H3Index cell, BBox *out, bool coverChildren) {
+    H3Error err;
+
+    CellBoundary boundary;
+    err = H3_EXPORT(cellToBoundary)(cell, &boundary);
+    if (err) {
+        return err;
+    }
+    // Convert to GeoLoop
+    GeoLoop loop;
+    loop.numVerts = boundary.numVerts;
+    loop.verts = (LatLng *)&boundary.verts;
+    // Calculate bbox
+    bboxFromGeoLoop(&loop, out);
+
+    if (coverChildren) {
+        // Buffer the bounding box to cover children
+        scaleBBox(out, CHILD_SCALE_FACTOR);
+    }
+
+    // Adjust the BBox to handle poles
+    H3Index poleTest;
+    // North pole
+    err = H3_EXPORT(latLngToCell)(&NORTH_POLE, H3_GET_RESOLUTION(cell),
+                                  &poleTest);
+    if (NEVER(err != E_SUCCESS)) {
+        return err;
+    }
```

Comment:
It looks like Clang warnings want these stores to be to different variables https://github.com/uber/h3/actions/runs/6408852991/job/17398764952?pr=785 rather than a reused `err` variable. Not 100% sure I agree (since this is really complaining about something that only affects the coverage build) although I think it is marginally clearer than reusing `err`.

Reply:
Ok, will update - I thought I was being clever by declaring this at the top of each function that could throw, but clearly not.